django/db/models/functions/base.py

```diff
@@ -203,8 +203,7 @@ def __init__(self, expression, substring, **extra):
         """
         if not hasattr(substring, 'resolve_expression'):
             substring = Value(substring)
```

Comment:
It looks like such simplification could be done in many places. But why `_output_field` and not `output_field`? `output_field` is documented for `Transform` (https://docs.djangoproject.com/en/1.11/ref/models/lookups/#django.db.models.Transform.output_field), maybe it should be documented for `Func` too?